### ISSUE TITLE
Android resource-id generated from testID not contains package name

### ISSUE THREAD

**Issue by pranshuchittora** [facebook/react-native#32237]
## Description
I am working on writing black-box E2E tests for React Native App.  
For targeting elements, I am using `testID` in JS world which translates to `resource-id` in `Android` and `name` in `iOS`.

The `resource-id` generated from `testID` is not prefixed with the name of the package. Due to which appium is not able to find elements by `resouce-id`. Works fine in `iOS`.

Current Behaviour
`testID = "abcd"` => `resource-id = "abcd"`

Expected
`testID = "abcd"` => `resource-id = "com.myawesomeapp:id/abcd"`

Refer this -> https://github.com/appium/appium/issues/15354#issuecomment-836679925

## React Native version:

```
System:
    OS: macOS 11.5.2
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 998.38 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.16.1 - ~/.nvm/versions/node/v14.16.1/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v14.16.1/bin/yarn
    npm: 6.14.12 - ~/.nvm/versions/node/v14.16.1/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    Android SDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.7199119
    Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.11 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.65.1 => 0.65.1 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```
## Steps To Reproduce

1. Define component
  ```jsx
  <Pressable testID="e2e_tid">
      <Text>Hello There</Text>
  </Pressable>
   
```
2. Build the app
3. Use `appium desktop` or any other tool to probe the render tree.  

## Expected Results

`testID` with explicitly defined prefix. (Expected)

```jsx
  <Pressable testID="com.rncliappium:id/e2e_tid">
      <Text>Hello There</Text>
  </Pressable>
```
https://user-images.githubusercontent.com/32242596/134005310-99a883b6-de31-4feb-ae50-2d0203a9b280.mov

`testID` without any prefix (Current Behaviour)

```jsx
  <Pressable testID="e2e_tid">
      <Text>Hello There</Text>
  </Pressable>
```
https://user-images.githubusercontent.com/32242596/134005523-53bef6ae-be02-489b-9cb5-881337ba8f2e.mov

**Comment by tuncaulubilge** [comment]
To add a bit of context, Android requires resource-ids to be unique amongst all packages, so it's best practice to prefix it with the bundle identifier. That's why Appium has opted in to use this prefix. See: https://github.com/appium/appium/issues/15354#issuecomment-865048105

For the time being, a solution like this could be used to manually add that prefix to every testID (see: https://dev.to/nextlevelbeard/an-end-to-the-abuse-of-accessibility-ids-5d2j), but we should really have this solution as part of React Native.

**Comment by keremoge** [comment]
Hi @pranshuchittora, Does the issue still persists? 
Can you please share your environment for Android? 
Appium with UIAutomator2 or Espresso?

**Comment by pranshuchittora** [comment]
> Hi @pranshuchittora, Does the issue still persists? Can you please share your environment for Android? Appium with UIAutomator2 or Espresso?

Hey @keremoge will test this out with the latest RN and latest Appium. Just give me a couple of days.

**Comment by BlackSwan2019** [comment]
Here's a workaround I used in my Appium scripts. You can cherry pick what is relevant for you (we use WebDriverIO + Cucumber).

In the WebDriverIO config file, in the `beforeCommand` hook, I intercept any commands that use `accessibility id` and change strategy to `xpath` with selector `//*[@resource-id=${<accessibility id>}]`. 

```
beforeCommand(commandName, args) {
     // Change "accessibility id" to "xpath" and select for "resource-id".
     if (driver.isAndroid && args.length === 2 && args[0] === 'accessibility id') {
          args[0] = 'xpath'
          args[1] = `//*[@resource-id="${args[1]}"]`
     }
}
```

`accessibility id` is same as `testID`. 

`xpath` gets a rap for being slower, but my tests are still running pretty fast, and the very short and generalized `xpath` makes it less fragile. 

Using the above workaround, I didn't have to add any helper methods to `testID` props (which would prefix every ID with my package name) to the thousands of `testID`s in our RN app. Not to mention that many of those `testID` props are passed down multiple custom components (to construct longer and more unique `testID`s), so we wouldn't want to call the helper method on every `testID` prop anyway.

**Comment by matsuda-argano** [comment]
@pranshuchittora 
I am very sorry that this question is not related to the Issue, but could you please tell me the name of the tool used in this capture?

<img width="538" alt="スクリーンショット 2022-12-19 17 10 56" src="https://user-images.githubusercontent.com/106219284/208377824-00507296-8fb0-483c-a849-6f565dd9dbdd.png">

**Comment by BlackSwan2019** [comment]
@matsuda-argano The tool he used is Appium Desktop. I recommend Appium Inspector though, if you already have Appium installed. It is the exact same tool as Appium Desktop, it just doesn't come bundled with Appium server, like Desktop does, so there is a little less bloat.
